README.md

```diff
@@ -1,41 +1,80 @@
-# Crate Name
+# vm-allocator
 
-## Design
+`vm-allocator` is a crate designed to to provide allocation and release strategies
```

Comment:
I think one relevant thing that is missing from this document is the design choice of having one allocator per resource. We should somehow write about this as well.